openedx/core/djangoapps/user_api/models.py

```diff
@@ -14,6 +14,37 @@
 from student.models import UserProfile, Registration, PendingEmailChange  # pylint: disable=unused-import
 
 
+class PreferenceRequestError(Exception):
+    """There was a problem with a preference request."""
+    pass
+
+
+class PreferenceNotFound(PreferenceRequestError):
+    """The desired user preference was not found."""
+    pass
+
+
+class PreferenceValidationError(PreferenceRequestError):
+    """
+    Validation issues were found with the supplied data. More detailed information is present
+    in preference_errors, a dict with specific information about each preference that failed
+    validation. For each preference, there will be at least a developer_message describing
+    the validation issue, and possibly also a user_message.
+    """
+    def __init__(self, preference_errors):
+        self.preference_errors = preference_errors
```

Comment:
I figured that they aren't actually fields so it seemed odd. Do you think of them as fields?